crates/snapbox/src/data.rs

```diff
@@ -112,37 +112,25 @@ impl Data {
     ///
     /// Note: this will **not** do a binary-content check
     pub fn make_text(&mut self) -> Result<(), std::str::Utf8Error> {
-        *self = Self::text(std::mem::take(self).into_string()?);
+        *self = Self::text(std::mem::take(self).try_string()?);
         Ok(())
     }
 
-    /// Coerce to a string
-    ///
-    /// Note: this will **not** do a binary-content check
-    pub fn into_string(self) -> Result<String, std::str::Utf8Error> {
-        match self.inner {
+    /// Tries to convert the underlying data into a String
+    pub fn try_string(&self) -> Result<String, std::str::Utf8Error> {
+        match &self.inner {
             DataInner::Binary(data) => {
-                let data = String::from_utf8(data).map_err(|e| e.utf8_error())?;
+                let data = String::from_utf8(data.clone()).map_err(|e| e.utf8_error())?;
                 Ok(data)
```

Comment:
Please revert this change.  We know the content is binary, we shouldn't be treating it as a string